### ISSUE TITLE
Meter

### ISSUE THREAD

**Issue by nerder** [buildo/react-components#258]
## Requirements

- name: Meter
- there should be a bar
  - should be filled based on the current value (always LTR)
  - minimum value could be <> 0
  - can be colored based on different values
    - but always a single color
- there should be a label
  - positioned right of the bar
  - could be a numeric (currentValue/max) or percentage value [different style]
  - can be colored based on different values

## Specs

- `<Meter>` should have a stateless interface and do not accept any child
- it have to take `current` in order to show the result and the progress
- each part of the component can be customizable in style and behaviour
  - label format and coloring
  - color of the filling
- it can be splitted in different steps. Each step accept a range value and can have a custom style as well

## API

```
<Meter 
  value={Number}
  min={Number}
  max={Number}
  labelFormatter={Function}
  steps={Array}
  baseFillingColor={String}
  baseLabelColor={String}
/>
```

## Defaults

**min**:  `0`
**max**:  `100`
**labelFormatter**:
```
labelFormatter = (current, min, max) => {
  return `${computeResult(current, min, max)}%`;
};
```
**steps**: 
```
[{ 
  startValue: min,
  endValue: max,
  fillingColor: defaultFillingColor,
  labelColor: defaultLabelColor
}]
```

**Comment by giogonzo** [comment]
as already discussed offline, I'm for `value`, `min` and `max` instead of `currentValue`, `minValue`, `maxValue` (don't add anything and are longer, plus value is sort of standard)

**Comment by nerder** [comment]
i'm a agree with this. I did it in the last commit. Specs are upgraded as well

**Comment by nerder** [comment]
### Proposal: Specs Editing

IMO is not explicit from the specs (even requirements actually) if our component can accept "holes" i mean that a gap between each step can be accepted or not, at the moment there is no checks for this, but a fallback on default is present instead this is just to don't break up the entire component. If we decided for not, is pretty obvious that `endValue` of each step will be equal of `startValue` of the next one. This can be very tedious and can let the user to make mistakes (that we have to check), if we precompute the ranges having just the `endValues` of each step, we can be sure to not make any hole, and reduce a lot the number of thing that we have to check (contiguity checks)

**Comment by lucacioria** [comment]
yes, I was wandering if/when this would be discussed :p 
It's what I meant when I was saying that this range representation is redundant, and @giogonzo said it was more natural. Either approach is valid, if justified.

During requirements gathering I said that the ranges must completely cover the full range [min,max] and I think we should stick to that.

**Comment by giogonzo** [comment]
> yes, I was wandering if/when this would be discussed :p 
It's what I meant when I was saying that this range representation is redundant, and @giogonzo said it was more natural. Either approach is valid, if justified.

In the end we decided that the ability to pass just `end` for each range (or symmetrical, just `start`) is a nice to have for the user and we could/should add it later on

> During requirements gathering I said that the ranges must completely cover the full range [min,max] and I think we should stick to that.

This has changed yesterday in our offline discussions: we decided that having a base style and special style modifiers for sparse ranges (still not overlapping) could be a reasonable feature to have, and it makes the component more generic. Requirements/specs are not updated yet 
@lucacioria can we briefly discuss this offline before proceeding?

/cc @FrancescoCioria
